internal/nginx/main.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nginx
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/tv42/httpunix"
+)
+
+// PID defines the location of the pid file used by NGINX
+var PID = "/tmp/nginx.pid"
+
+// StatusSocket defines the location of the unix socket used by NGINX for the status server
+var StatusSocket = "/tmp/nginx-status-server.sock"
+
+// HealthPath defines the path used to define the health check location in NGINX
+var HealthPath = "/healthz"
+
+// StatusPath defines the path used to expose the NGINX status page
+// http://nginx.org/en/docs/http/ngx_http_stub_status_module.html
+var StatusPath = "/nginx_status"
+
+// StreamSocket defines the location of the unix socket used by NGINX for the NGINX stream configuration socket
+var StreamSocket = "/tmp/ingress-stream.sock"
+
+var nginxUnixLocation = "nginx-status"
```

Comment:
nitpick: `statusLocation` probably makes more sense since this is an alias to `StatusSocket`

Reply:
done